File: rdf4j-java/src/main/java/eu/neverblink/jelly/convert/rdf4j/rio/JellyParserFactory.java
package eu.neverblink.jelly.convert.rdf4j.rio;

import static eu.neverblink.jelly.convert.rdf4j.rio.JellyFormat.JELLY;

import eu.neverblink.jelly.convert.rdf4j.Rdf4jConverterFactory;
import org.eclipse.rdf4j.rio.RDFFormat;
import org.eclipse.rdf4j.rio.RDFParser;
import org.eclipse.rdf4j.rio.RDFParserFactory;

public final class JellyParserFactory implements RDFParserFactory {

    @Override
    public RDFFormat getRDFFormat() {
        return JELLY;
    }

    @Override
    public RDFParser getParser() {
        final var converterFactory = Rdf4jConverterFactory.getInstance();
        return new JellyParser(converterFactory);
    }
}

File: rdf4j/src/main/scala/eu/ostrzyciel/jelly/convert/rdf4j/rio/JellyParserFactory.scala
package eu.ostrzyciel.jelly.convert.rdf4j.rio

import org.eclipse.rdf4j.rio.{RDFFormat, RDFParserFactory}

final class JellyParserFactory extends RDFParserFactory:
  override def getRDFFormat: RDFFormat = JELLY

  override def getParser: JellyParser = JellyParser()
